### PR TITLE
chore: rename OrderableList to LegacyOrderableList

### PR DESCRIPTION
**Pull request by anxobotana**
In order to replace `react-dnd` we need to replace OrderableList, the only component using it.
First step use LegacyOrderableList, this should work the same way as OrderableList

### CODE REVIEW

File: src/components/LegacyOrderableList/LegacyOrderableList.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        const renderWithDragState = (_item: TestItem, { componentDragState }: LegacyDragProperties) => (
+            <div style={{ height: `${ITEM_HEIGHT}px` }}>{componentDragState}</div>
+        );
+
+        cy.mount(<OrderableListWithDefaultProps renderContent={renderWithDragState} />);
+
+        cy.get(DRAGGABLE_ITEM).each(($el) => {
+            const dataTransfer = new DataTransfer();
+            cy.wrap($el)
+                .should('have.text', LegacyLegacyItemDragState.Idle)
```
Comment (imoutaharik) [inline src/components/LegacyOrderableList/LegacyOrderableList.cy.tsx:125]: Got a Legacy too many here.

File: src/components/LegacyOrderableList/LegacyOrderableList.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { Meta, StoryFn } from '@storybook/react';
+import { useState } from 'react';
+import { LegacyOrderableList as OrderableListComponent } from './LegacyOrderableList';
+import { LegacyOrderableListItem } from './types';
+import { LegacyOrderableListProps } from '.';
+import { chain } from '@react-aria/utils';
+import { renderContent, storyItems } from '@components/LegacyOrderableList/utils';
+
+export default {
+    title: 'Components/Legacy Orderable List',
```
Comment (imoutaharik) [inline src/components/LegacyOrderableList/LegacyOrderableList.stories.tsx:12]: You could move this to `Deprecated` category in Storybook -> `Deprecated/Legacy OrderableList`

File: src/components/index.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ export * from './FrontifyPattern';
 export * from './Grid';
 export * from './InlineDialog';
 export * from './InputLabel';
+export * from './LegacyOrderableList';
```
Comment (imoutaharik) [inline src/components/index.ts:35]: Did you just copy OrderableList and created the Legacy?
Are you keeping the old one as `OrderableList` for later refactoring ?

**Review by imoutaharik — COMMENTED** [review]
There is only 3 instances of that component in web-app that could be renamed pretty quickly. But before refactoring the whole component we should make sure there is no impact in other repositories that uses Fondue like `guidelines-block`.

File: src/components/index.ts (hunk 3 above)
Comment (anxobotana) [reply src/components/index.ts:35]: Yup, the idea is having a new `OrderableList` using `Tree` component.
In fact i have already the new component almost ready

**Review by imoutaharik — APPROVED** [review]

**Comment by anxobotana** [comment]
@imoutaharik can you merge this PR? 
SonarCloud complains about duplication are intended, LegacyOrderableList is a copy of OrderableList, a follow up PR will implement the new one

**Pull request merged**